### PR TITLE
Add analytics on the CLI

### PR DESCRIPTION
**Pull request by antho1404**

### CODE REVIEW

File: packages/cli/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 - [#97](https://github.com/liteflow-labs/liteflow-js/pull/97) Use the new pre-configured developer image of `mesg/engine`.
 - [#103](https://github.com/liteflow-labs/liteflow-js/pull/103) Switch to engine v0.24 with better dev environment.
 - [#106](https://github.com/liteflow-labs/liteflow-js/pull/106) Simplify service template.
+- [#](https://github.com/liteflow-labs/liteflow-js/pull/) Add analytics for the commands.
```
Comment (antho1404) [inline packages/cli/CHANGELOG.md:22]:
```suggestion
- [#109](https://github.com/liteflow-labs/liteflow-js/pull/109) Add analytics for the commands.
```

File: packages/cli/src/utils/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  const filePath = join(command.config.configDir, 'analytics')
+  if (!existsSync(filePath)) {
+    writeFileSync(filePath, uuid.v4())
+  }
+  const user = readFileSync(filePath).toString()
+  const visitor = ua('UA-164307780-2', user, {
+    strictCidFormat: false,
+    https: true
+  })
+  visitor.set('cd1', process.platform) // platform
```
Comment (NicolasMahe) [inline packages/cli/src/utils/analytics.ts:17]: why cdX?

File: packages/cli/src/utils/analytics.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import Command from '@oclif/command'
+import { existsSync, writeFileSync, readFileSync } from 'fs-extra'
+import { join } from 'path'
+
+export default (command: Command, event: string, label?: string, value?: string | number) => {
+  const filePath = join(command.config.configDir, 'analytics')
+  if (!existsSync(filePath)) {
+    writeFileSync(filePath, uuid.v4())
+  }
+  const user = readFileSync(filePath).toString()
```
Comment (NicolasMahe) [inline packages/cli/src/utils/analytics.ts:12]: can't we use the user id when logged?
```
var visitor = ua('UA-XXXX-XX', {uid: 'as8eknlll'});
```
https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#uid

File: packages/cli/src/commands/service/init.ts [hunk 4]
```diff
@@ -20,6 +21,7 @@ export default class Init extends Command {
 
   async run() {
     const { args, flags } = this.parse(Init)
+    await analytics(this, 'start', 'dir', args.DIR)
```
Comment (NicolasMahe) [inline packages/cli/src/commands/service/init.ts:24]: do we need to wait for the request?
let's implement it as a send and forget
```suggestion
    analytics(this, 'start', 'dir', args.DIR)
```

File: packages/cli/src/commands/service/execute.ts [hunk 5]
```diff
@@ -127,6 +129,7 @@ export default class Execute extends Command {
     await tasks.run()
 
     styledJSON(decode(execution.outputs))
+    await analytics(this, 'end')
```
Comment (NicolasMahe) [inline packages/cli/src/commands/service/execute.ts:132]: is it useful to track the end of the command?
just one event by command should be enough

File: packages/cli/src/utils/analytics.ts (hunk 2 above)
Comment (antho1404) [reply packages/cli/src/utils/analytics.ts:17]: this is to have custom metrics so like that we can identify the environment of the user and select all user that run the dev command on mac for exemple

File: packages/cli/src/commands/service/execute.ts (hunk 5 above)
Comment (antho1404) [reply packages/cli/src/commands/service/execute.ts]: removed

File: packages/cli/src/commands/service/init.ts (hunk 4 above)
Comment (antho1404) [reply packages/cli/src/commands/service/init.ts]: done

File: packages/cli/src/utils/analytics.ts (hunk 3 above)
Comment (antho1404) [reply packages/cli/src/utils/analytics.ts]: Yes definitely, for now it is not possible as we don't store this info but yes definitely agree that this should be improved

**Pull request merged**